test/handshake_test.cpp

```diff
@@ -27,6 +27,85 @@ TEST_CASE("handshake") {
 
   client_stream.next_layer().connect(server_stream.next_layer());
 
+  SECTION("invalid certificate data") {
+    boost::system::error_code error{};
+    const std::string bad_cert = "DECAFBAD";
+    client_ctx.add_certificate_authority(net::buffer(bad_cert), error);
+    CHECK(error.category() == boost::system::system_category());
+    CHECK(error.value() == ERROR_INVALID_DATA);
+  }
+
+  SECTION("no certificate validation") {
+    boost::system::error_code verify_error{};
+    client_ctx.set_verify_mode(boost::windows_sspi::verify_none, verify_error);
+    REQUIRE_FALSE(verify_error);
+
+    boost::system::error_code client_error{};
+    client_stream.async_handshake(boost::windows_sspi::stream_base::client,
+                                  [&client_error, &io_context](const boost::system::error_code& ec) {
+                                    client_error = ec;
+                                    io_context.stop();
+                                  });
+
+    boost::system::error_code server_error{};
+    server_stream.async_handshake(boost::asio::ssl::stream_base::server,
+                                  [&server_error](const boost::system::error_code& ec) {
+                                    server_error = ec;
+                                  });
+    io_context.run();
+    CHECK_FALSE(client_error);
```

Comment:
You might consider initializing the error_code to an error value, because an empty implementation of `async_handshake` would pass this test :).

Reply:
:-) Good point. Will do.